packages/account/src/Sections/Profile/PhoneVerification/resend-code-timer.tsx

```diff
@@ -1,64 +1,119 @@
 import React from 'react';
 import { Button, CaptionText } from '@deriv-com/quill-ui';
 import { Localize } from '@deriv/translations';
-import { useVerifyEmail } from '@deriv/hooks';
+import { useSettings, useVerifyEmail } from '@deriv/hooks';
+import dayjs from 'dayjs';
 
 type TResendCodeTimer = {
-    resend_code_text: string;
-    count_from: number;
+    is_button_disabled: boolean;
+    should_show_resend_code_button: boolean;
+    setIsButtonDisabled: (value: boolean) => void;
     setShouldShowDidntGetTheCodeModal: (value: boolean) => void;
-    start_timer: boolean;
-    setStartTimer: (value: boolean) => void;
+    reInitializeGetSettings: () => void;
 };
 const ResendCodeTimer = ({
-    count_from = 60,
-    resend_code_text,
+    is_button_disabled,
+    should_show_resend_code_button,
+    setIsButtonDisabled,
     setShouldShowDidntGetTheCodeModal,
-    start_timer,
-    setStartTimer,
+    reInitializeGetSettings,
 }: TResendCodeTimer) => {
-    // TODO: calculate count_from and time units(secs or mins) using timestamp once mockApi is finalised
-    // TODO: revist start timer logic and localizing the title
-    const [timer, setTimer] = React.useState(count_from);
-    const { send } = useVerifyEmail('phone_number_verification');
-    const initial_timer_title =
-        resend_code_text === 'Resend code' ? `Resend code in ${timer}s` : `Didn’t get the code? (${timer}s)`;
-    const [timer_title, setTimerTitle] = React.useState(initial_timer_title);
+    const [timer, setTimer] = React.useState<number | undefined>();
+    const [next_otp_request, setNextOtpRequest] = React.useState('');
+    // @ts-expect-error this for now
+    const { send, is_success } = useVerifyEmail('phone_number_verification');
+    const { data: account_settings } = useSettings();
+    const current_time = dayjs();
 
-    const setTitle = (timer: number, text: string) => {
-        const title = text === 'Resend code' ? `Resend code in ${timer}s` : `Didn’t get the code? (${timer}s)`;
-        setTimerTitle(title);
-    };
+    const setTitle = React.useCallback(
+        (timer: number) => {
+            let display_time: string;
+            if (timer > 60) {
+                display_time = `${timer / 60}m`;
+            } else {
+                display_time = `${timer}s`;
+            }
+            should_show_resend_code_button
+                ? setNextOtpRequest(` in ${display_time}`)
+                : setNextOtpRequest(` (${display_time})`);
+        },
+        [should_show_resend_code_button]
+    );
+
+    React.useEffect(() => {
+        if (is_success) reInitializeGetSettings();
```

Comment:
do we reinitialise this call to get recent timestamp?

Reply:
Yes we need to reinitialise the `get_settings` call after user finish sending a otp request. By default it doesn't retrigger the get settings call which will not run the timer. By includding this, it will calls get_settings again to trace the timestamp coming from BE